### Hi Eveyrone
Welcome to my first data analysis studies
<br>I'll be analyzing data from the Google Play Store in this notebook. I applied Python to the data analysis.
<br>Reviews of apps available on Google Play There are numerous mobile apps. They can be profitable and are simple to make. These two reasons are driving the development of an increasing number of apps. In this notebook, we'll compare over 10,000 apps from different categories on Google Play to conduct a thorough analysis of the Android app industry. In order to create strategies to promote growth and retention, we will look for insights in the data.
<br>The data we are going to analyze consists of the following two files:

* **playstore data.csv:** provides all of the information on the Google Play applications. Each specific app is described by 13 features.
* **user_reviews.csv:** provides reviews of every app. Three new features, Sentiment (Positive, Negative, or Neutral), Sentiment Polarity, and Sentiment Subjectivity, have been pre-processed into the text from each review.

I'll first go over the EDA analysis before getting into the given data.

# **Problem Statements**
1. What are the top categories on Play Store?
2. Are majority of the apps Paid or Free?
3. How importance is the rating of the application?
4. Which categories from the audience should the app be based on?
5. Which category has the most no. of installations?
6. How does the count of apps varies by Genres?
7. How does the last update has an effect on the rating?
8. How are ratings affected when the app is a paid one?
9. How are reviews and ratings co-related?
10. Lets us discuss the sentiment subjectivity.
11. Is subjectivity and polarity proportional to each other?
12. What is the percentage of review sentiments?
13. How is sentiment polarity varying for paid and free apps?
14. How Content Rating affect over the App?
15. Does Last Update date has an effects on rating?
16. Distribution of App update over the Year.
17. Distribution of Paid and Free app updated over the Month.

# **What is Exploratory Data Analysis?**
Exploratory data analysis (EDA) is used by data scientists to examine data sets for patterns and anomalies (outliers), to create hypotheses based on our comprehension of the dataset, and to summarize their key properties. Data visualization techniques are frequently utilized in this process. In any project involving data analysis or data science, it is a crucial stage. It helps in deciding how to best manipulate data sources to obtain the desired results.

In order to better comprehend the data and make it more visually attractive and appealing, EDA involves producing summary statistics for the dataset's numerical data.

The following are the various steps involved in the EDA process:
1. <b>Problem Statement</b> - We'll come up with ideas and analyze the provided data. We will investigate the characteristics it has and attempt to conduct a philosophical examination of their significance and relevance to this issue.
2. <b>Hypothesis</b> - Following our examination of the data base's features, we will formulate some fundamental hypotheses on which to base our exploration of the data's potential outcomes.
3. <b>Univariate Analysis</b> - It is the most straightforward method of data analysis. In order to start, we would choose just one attribute and thoroughly examine it. It deals with no co-relation whatsoever, and its primary goal is to describe. Data is taken, summarized, and patterns are discovered within the data.
4. <b>Bivariate Analysis</b> - Cause and the connection between the two traits are the subjects of this analysis. We'll make an effort to comprehend how traits are related to one another.
5. <b>Multivariate Analysis</b> - this is what's done.when it's necessary to investigate multiple variables at once.
5. <b>Data Cleaning</b> - The dataset will be cleaned, and missing data, outliers, and categorical variables will be handled.
6. <b>Testing Hypothesis</b> - We will determine whether our data complies with the assumptions needed by the majority of multivariate approaches.

# **Exploring Play store data:**

In [6]:
# Importing libraries
import pandas as pd # data processing, CSV file I/O (e.g. pd.read_csv)
import numpy as np # linear algebra
import matplotlib.pyplot as plt
import seaborn as sns  # visualization tool
from datetime import datetime
# plotly
#import plotly 
#plotly.offline.init_notebook_mode(connected=True)
#import plotly.graph_objs as go
import warnings
#sns.set(font_scale=1.5)
warnings.filterwarnings("ignore")

# **Lets Explore play store DataFrame**


In [8]:
# loading csv File
df = pd.read_csv('googleplaystore.csv')

In [9]:
# Display the Play Store App data
play_store=pd.concat([df.head(),df.tail()])
play_store

App             Category  \
0         Photo Editor & Candy Camera & Grid & ScrapBook       ART_AND_DESIGN   
1                                    Coloring book moana       ART_AND_DESIGN   
2      U Launcher Lite – FREE Live Cool Themes, Hide ...       ART_AND_DESIGN   
3                                  Sketch - Draw & Paint       ART_AND_DESIGN   
4                  Pixel Draw - Number Art Coloring Book       ART_AND_DESIGN   
10836                                   Sya9a Maroc - FR               FAMILY   
10837                   Fr. Mike Schmitz Audio Teachings               FAMILY   
10838                             Parkinson Exercices FR              MEDICAL   
10839                      The SCP Foundation DB fr nn5n  BOOKS_AND_REFERENCE   
10840      iHoroscope - 2018 Daily Horoscope & Astrology            LIFESTYLE   

       Rating Reviews                Size     Installs  Type Price  \
0         4.1     159                 19M      10,000+  Free     0   
1         3.9     967                 14M     500,000+  Free     0   
2         4.7   87510                8.7M   5,000,000+  Free     0   
3         4.5  215644                 25M  50,000,000+  Free     0   
4         4.3     967                2.8M     100,000+  Free     0   
10836     4.5      38                 53M       5,000+  Free     0   
10837     5.0       4                3.6M         100+  Free     0   
10838     NaN       3                9.5M       1,000+  Free     0   
10839     4.5     114  Varies with device       1,000+  Free     0   
10840     4.5  398307                 19M  10,000,000+  Free     0   

      Content Rating                     Genres      Last Updated  \
0           Everyone               Art & Design   January 7, 2018   
1           Everyone  Art & Design;Pretend Play  January 15, 2018   
2           Everyone               Art & Design    August 1, 2018   
3               Teen               Art & Design      June 8, 2018   
4           Everyone    Art & Design;Creativity     June 20, 2018   
10836       Everyone                  Education     July 25, 2017   
10837       Everyone                  Education      July 6, 2018   
10838       Everyone                    Medical  January 20, 2017   
10839     Mature 17+          Books & Reference  January 19, 2015   
10840       Everyone                  Lifestyle     July 25, 2018   

              Current Ver         Android Ver  
0                   1.0.0        4.0.3 and up  
1                   2.0.0        4.0.3 and up  
2                   1.2.4        4.0.3 and up  
3      Varies with device          4.2 and up  
4                     1.1          4.4 and up  
10836                1.48          4.1 and up  
10837                 1.0          4.1 and up  
10838                 1.0          2.2 and up  
10839  Varies with device  Varies with device  
10840  Varies with device  Varies with device

In [10]:
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 10841 entries, 0 to 10840
Data columns (total 13 columns):
 #   Column          Non-Null Count  Dtype  
---  ------          --------------  -----  
 0   App             10841 non-null  object 
 1   Category        10841 non-null  object 
 2   Rating          9367 non-null   float64
 3   Reviews         10841 non-null  object 
 4   Size            10841 non-null  object 
 5   Installs        10841 non-null  object 
 6   Type            10840 non-null  object 
 7   Price           10841 non-null  object 
 8   Content Rating  10840 non-null  object 
 9   Genres          10841 non-null  object 
 10  Last Updated    10841 non-null  object 
 11  Current Ver     10833 non-null  object 
 12  Android Ver     10838 non-null  object 
dtypes: float64(1), object(12)
memory usage: 1.1+ MB


In [13]:
# Finding the number of rows and columns in the given dataset
print(df.columns)
rows=df.shape[0]
columns=df.shape[1]
print(f"the no of rows is {rows} and no of columns is {columns}")

Index(['App', 'Category', 'Rating', 'Reviews', 'Size', 'Installs', 'Type',
       'Price', 'Content Rating', 'Genres', 'Last Updated', 'Current Ver',
       'Android Ver'],
      dtype='object')
the no of rows is 10841 and no of columns is 13


In [14]:
df.describe()

Rating
count  9367.000000
mean      4.193338
std       0.537431
min       1.000000
25%       4.000000
50%       4.300000
75%       4.500000
max      19.000000

**Let us first define what information the columns contain based on our inspection.**

play_store dataframe has 10841 rows and 13 columns. The 13 columns are identified as below:
1. **App** - It tells us about the name of the application with a short description (optional).
2. **Category** - It gives the category to the app.
3. **Rating** - It contains the average rating the respective app received from its users.
4. **Reviews** - It tells us about the total number of users who have given a review for the application.
5. **Size** - It tells us about the size being occupied the application on the mobile phone.
6. **Installs** - It tells us about the total number of installs/downloads for an application.
7. **Type** - IIt states whether an app is free to use or paid.
8. **Price** - It gives the price payable to install the app. For free type apps, the price is zero.
9. **Content Rating** - It states whether or not an app is suitable for all age groups or not.
10. **Genres** - It tells us about the various other categories to which an application can belong.
11. **Last Updated** - It tells us about the when the application was updated.
12. **Current Ver** - It tells us about the current version of the application.
13.**Android Ver** - It tells us about the android version which can support the application on its platform.